### ISSUE TITLE
scikit-misc fails to build with Python 3.12.0a4

### ISSUE THREAD

**Issue by penguinpee** [has2k1/scikit-misc#20]
Fedora is preparing for the inclusion of Python 3.12 in Fedora 39, planned to be released at the end of October this year. Currently scikit-misc fails to build with Python 3.12 due to deprecations:

```python
Traceback (most recent call last):
  File "/usr/lib/rpm/redhat/pyproject_buildrequires.py", line 508, in main
    generate_requires(
  File "/usr/lib/rpm/redhat/pyproject_buildrequires.py", line 415, in generate_requires
    generate_build_requirements(backend, requirements)
  File "/usr/lib/rpm/redhat/pyproject_buildrequires.py", line 265, in generate_build_requirements
    new_reqs = get_requires()
               ^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
    return self._get_build_requires(config_settings, requirements=['wheel'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
    self.run_setup()
  File "/usr/lib/python3.12/site-packages/setuptools/build_meta.py", line 484, in run_setup
    self).run_setup(setup_script=setup_script)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/setuptools/build_meta.py", line 335, in run_setup
    exec(code, locals())
  File "<string>", line 170, in <module>
  File "<string>", line 152, in setup_package
  File "/builddir/build/BUILD/scikit-misc-0.1.4/versioneer.py", line 1476, in get_version
    return get_versions()["version"]
           ^^^^^^^^^^^^^^
  File "/builddir/build/BUILD/scikit-misc-0.1.4/versioneer.py", line 1408, in get_versions
    cfg = get_config_from_root(root)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builddir/build/BUILD/scikit-misc-0.1.4/versioneer.py", line 342, in get_config_from_root
    parser = configparser.SafeConfigParser()
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
```

More information can be found in the downstream bug report:

https://bugzilla.redhat.com/show_bug.cgi?id=2165562

**Comment by has2k1** [comment]
I have released v0.2.0 which uses [meson-build](https://mesonbuild.com). `numpy.distutils` is no more and overall scikit-misc should be easier to package.

**Comment by penguinpee** [comment]
That's solved. But I just hit a rock trying to build 0.3.0. I'll open a new issue.
